core/src/main/clojure/xtdb/sql/plan.clj

```diff
@@ -138,6 +138,112 @@
                             {::err/message (str "Cannot build interval for: "  (pr-str qualifier))
                              :qualifier qualifier}))))
 
+(defn interval-qualifier->map [qualifier]
+  (r/zmatch qualifier
+            [:interval_qualifier
```

Comment:
wonder whether we can avoid the cross-product of different possibilities here with a couple of extracted fns for start/end?